mlflow/utils/uri.py

```diff
@@ -486,8 +486,9 @@ def _decode(url):
     # Keep decoding until the url stops changing (with a max of 10 iterations)
     for _ in range(10):
```

Comment:
i think we can consider removing this loop, otherwise someone can just construct a malicious query string that requires 11 rounds of decoding. 

i saw the [original discussion on the PR](https://github.com/mlflow/mlflow/pull/10653#discussion_r1419996163), but i feel like there isn't much of a chance that an infinite loop will happen, given that there's only two things that can happen:

1. If the url has decodable parts, then it will get smaller (i.e. less url-decodable)
2. If it has no decodable parts, then we return

Reply:
Sounds good. Let's remove it

Reply:
oh wait sorry, i missed that we throw after 10 iterations, i think it's fine to leave it in actually

Reply:
How about using a library function for canonicalising URLs from a package like [W3lib](https://pypi.org/project/w3lib/) or similar? While that might add a dependency, it could be safer than something written ad-hoc.

Reply:
@yitzikc using a library makes sense, i think this will probably be done in a future PR though so we can have some time to evaluate it